File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = -1
    
    citations.sort(reverse = True)
    len_c = len(citations)
    
    if citations[0] == 0 :
        return 0
    
    for i in range(len_c) :
        cnt = i + 1
        value = citations[i]
        
        answer = max(answer, min(cnt,value))
    
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        
        List<Integer> cit_list = new ArrayList<>();
        for (int c : citations) {
            cit_list.add(c);
        }
        Collections.sort(cit_list, Collections.reverseOrder());

        for (int i=0; i<citations.length; i++) {
            if (i+1 <= cit_list.get(i)) {
                answer = i+1;
            } else {
                break;
            }
        }
        
        return answer;
    }
}